utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java

```diff
@@ -335,98 +445,11 @@ private static DescriptorProto messageElementToDescriptorProto(
                     .setMapEntry(isMapEntry);
             message.protoBuilder().mergeOptions(optionsBuilder.build());
         }
+        message.protoBuilder().addAllNestedType(allNestedTypes.values());
```

Comment:
If only values are being stored, do we need a map of locations? Are keys being used anywhere?

Reply:
NestedTypes cannot be simplified as List due to that the Location information is not available in the DescriptorProto so that we need to store it as the key.